### PR TITLE
instance parameters only appended to URL when appropriate

### PR DESCRIPTION
**Pull request by ChemiKyle**
Addresses issue #67

An `instance` parameter will only be added to `next_step_path` if ALL criteria are met:

- The user is on a repeating instrument
- The next instrument is within the same event and is also a repeating instrument

~~A possible improvement may be to detect if the next instrument has any existing instances before appending the `instance` parameter.~~ This is not how repeating forms are expected to work.

___
UPDATE: 

An `instance` parameter will only be added to `next_step_path` if ALL criteria are met:

- The user is on form in a repeating _event_
- They are already working on an instance > 1

### CODE REVIEW

**Comment by pbchase** [comment]
here's a test project you might like. 
[FRSLTest2WithAllRepe_2020-03-11_1709.REDCap.xml.zip](https://github.com/ctsit/form_render_skip_logic/files/4320936/FRSLTest2WithAllRepe_2020-03-11_1709.REDCap.xml.zip)

**Comment by ChemiKyle** [comment]
This PR as-is actually reintroduces issue #41 while solving #67, I am working on refactoring to appease both and will likely submit as a new PR entirely.

File: ExternalModule.php [hunk 1]
```diff
@@ -415,7 +415,14 @@ protected function loadFRSL($location, $record = null, $event_id = null, $instru
 
             if (isset($next_instrument)) {
                 // Path to the next available form in the current event.
-                $next_step_path = APP_PATH_WEBROOT . 'DataEntry/index.php?pid=' . $Proj->project_id . '&id=' . $record . '&event_id=' . $event_id . '&page=' . $next_instrument . '&instance=' . $instance;
+                $next_step_path = APP_PATH_WEBROOT . 'DataEntry/index.php?pid=' . $Proj->project_id . '&id=' . $record . '&event_id=' . $event_id . '&page=' . $next_instrument;
+
+                // If this is a repeating event, maintain the instance
+                if ($Proj->hasRepeatingFormsEvents() && $instance) {
+                    if ($Proj->RepeatingFormsEvents[$event_id] == "WHOLE") {
```
Comment (ChemiKyle) [inline ExternalModule.php]: Note to self: the following function call is not suitable; it results in a false positive if an event has all forms repeating but the event itself is not defined as repeating.

```php
if( Proj->allEventFormsAreRepeatingForms($event_id) ) {
```

**Comment by pbchase** [comment]
Tested in REDCap 9.7.0 and 8.10.1 with the two test projects in 62b7fde

**Comment by mlurca** [comment]
Thank you everyone for fixing this

**Pull request merged**